File: Lab-06-solution/src/test/java/com/petclinic/core/VisitServiceUnitTest.java
package com.petclinic.core;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import java.time.LocalDate;

import static org.assertj.core.api.Assertions.*;
import static org.mockito.Mockito.when;

@ExtendWith(MockitoExtension.class)
class VisitServiceUnitTest {
    @Mock
    private VisitRepository visitRepository;
    @InjectMocks
    private VisitService visitService;

    @Test
    public void shouldFindVisitByReferenceNumber() {
        var visit = new Visit(1L, "V01-23", LocalDate.of(2013, 12, 21), "Teeth whitening");
        when(this.visitRepository.findByReferenceNumber("V01-23")).thenReturn(visit);
        assertThat(this.visitService.findByReferenceNumber("V01-23").getReferenceNumber()).isEqualTo("V01-23");
    }

}

File: Lab-kotlin/src/test/kotlin/com/petclinic/core/VisitServiceUnitTest.kt
package com.petclinic.core

import org.assertj.core.api.Assertions.*
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.mockito.Mockito
import org.mockito.junit.jupiter.MockitoExtension
import java.time.LocalDate
import java.util.*

@ExtendWith(MockitoExtension::class)
internal class VisitServiceUnitTest {
    private val visitRepository = Mockito.mock(VisitRepository::class.java)
    private val visitService = VisitService(visitRepository)

    @Test
    fun shouldFindVisitByReferenceNumber() {
        val visit = Visit(1, LocalDate.now(), "teeth whitening", null, null)
        Mockito.`when`(visitRepository.findById(1)).thenReturn(Optional.of(visit))

        val foundVisit = visitService.findById(1)

        assertThat(foundVisit.id).isEqualTo(1)
    }
}
